sources/packages/backend/apps/api/src/route-controllers/designation-agreement/designation-agreement.institutions.controller.ts

```diff
@@ -89,13 +93,40 @@ export class DesignationAgreementInstitutionsController extends BaseController {
       );
     }
 
+    // Validate that all locations in the payload are authorized for the user.
```

Comment:
IMO, we are not asserting authorization here, but if the numbers received are valid IDs under the institution.